lms/resources/_js_config.py

```diff
@@ -159,15 +170,28 @@ def enable_content_item_selection_mode(self, form_action, form_fields):
         :param form_fields: the fields (keys and values) to include in the
             HTML form that we'll use to submit the user's chosen document
         :type form_fields: dict
+
+        :raise ConsumerKeyError: if request.lti_user.consumer_key isn't in the DB
         """
         self._config["mode"] = "content-item-selection"
 
         def google_picker_origin():
+            """
+            Return the URL of the top-most page that the LMS app is running in.
+
+            The frontend has to pass this to Google Picker, otherwise Google
+            Picker refuses to launch in an iframe.
+
+            :raise ConsumerKeyError: if request.lti_user.consumer_key isn't in the DB
+            """
             # Pass the URL of the LMS that is launching us to our JavaScript code.
             # When we're being launched in an iframe within the LMS our JavaScript
             # needs to pass this URL (which is the URL of the top-most page) to Google
             # Picker, otherwise Picker refuses to launch inside an iframe.
-            return self._context.custom_canvas_api_domain or self._ai_getter.lms_url()
+            return (
+                self._context.custom_canvas_api_domain
+                or self._application_instance().lms_url
+            )
```

Comment:
This is simply equivalent to the previous code and, like the previous code, it can raise `ConsumerKeyError` (which I added to the docstring)

Reply:
This PR + review is getting mega huge, so probably take this suggestion as an idea for a follow up (it would be nice to get this one done) but this feels very extra to me to define a function which just returns a single value and call it later _and_ to document it too. Could we put this in a variable instead, or just put the value inline later on?

```suggestion
google_picker_origin = (
    self._context.custom_canvas_api_domain
    or self._application_instance().lms_url
)
```